app/src/main/java/org/wikipedia/dataclient/Service.java

```diff
@@ -336,8 +336,8 @@
     @GET(MW_API_PREFIX + "action=query&prop=info&converttitles=&redirects=&inprop=watched")
     @NonNull Observable<MwQueryResponse> getWatchedInfo(@NonNull @Query("titles") String titles);
 
-    @GET(MW_API_PREFIX + "action=query&meta=siteinfo&list=watchlist&wllimit=500&wlallrev=1&wlprop=ids|title|flags|comment|parsedcomment|timestamp|sizes|user")
-    @NonNull Observable<MwQueryResponse> getWatchlist();
+    @GET(MW_API_PREFIX + "action=query&meta=allmessages&list=watchlist&wllimit=500&wlallrev=1&wlprop=ids|title|flags|comment|parsedcomment|timestamp|sizes|user")
+    @NonNull Observable<MwQueryResponse> getWatchlist(@NonNull @Query("ammessages") String languageCode);
```

Comment:
I'm not quite seeing what this is trying to do. If you pass just a language code to the `allmessages` API, it won't return anything meaningful, and will just cause load on the server.  Could you clarify what the goal is here?

Reply:
Make senses. I change the logic that creates a temporary list to store enabled language codes and it works as expected now.